src/Options/Commander.hs

```diff
@@ -181,10 +183,20 @@ unrenderSmall = flip Prelude.lookup [(pack $ show x, x) | x <- [minBound..maxBou
 instance Unrender () where
   unrender = unrenderSmall
 
-instance (Unrender a, Unrender b) => Unrender (Either a b) where
-  unrender x = leftCase x <|> rightCase x where
-    leftCase  = fmap Left  . unrender
-    rightCase = fmap Right . unrender
+-- | Allows alternate unrendering.
+data Alternate a b = Alternate a | Primary b deriving (Show, Eq, Ord, Functor)
+
+-- | Compatibility type for the old Unrender Either instance. Use altEither to convert AltEither to Either.
+type AltEither a b = Alternate b a
+
+-- | Compatibility type for the old Unrender Either instance.
+altEither :: forall a b. AltEither a b -> Either a b
+altEither = \case
+  Alternate x -> Right x
+  Primary   x -> Left  x
```

Comment:
`Compatibility function for the old Unrender Either instance` is preferred. Also, I would prefer to put this in an `Options.Commander.Compat` module and deprecating it, noting that it will be removed on a future major release and informing the user of the migration path.

Another path that requires no deprecation would be to add a combinator `alternatively`, mirroring `either`, using which one can easily inline `altEither`. Then add an instance for `Either` with a custom type error explaining the migration path. I think I slightly prefer the latter path, but I'll leave it up to you as the primary contributor as of late.